### PR TITLE
Tag Picker: fix select all and delete

### PR DESCRIPTION
**Pull request by micahgodbolt**
This PR starts to address the issues in #24314, fixing a bug that broke ctrl+a selection, and 'del' key deletion. This PR also fixes the styles so that they are properly driven from the is-selected class, and are applied to all items when ctrl + a is pressed.

After this PR I'll come back and look at adding click to select, which will be a bit more complicated and more of a feature add.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/4de0899dd07982aabdc611b1837bf9acfca44efc'>4de0899dd07982aabdc611b1837bf9acfca44efc</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react/src/components/pickers/TagPicker/TagPicker.tsx [hunk 1]
```diff
@@ -18,7 +18,6 @@ export class TagPickerBase extends BasePicker<ITag, ITagPickerProps> {
 
   constructor(props: ITagPickerProps) {
     super(props);
-    initializeComponentRef(this);
```
Comment (smhigley) [inline packages/react/src/components/pickers/TagPicker/TagPicker.tsx:21]: if the only thing the constructor is doing is calling `super`, should we just take it out entirely?

File: packages/react/src/components/pickers/TagPicker/TagItem.styles.ts [hunk 2]
```diff
@@ -103,6 +104,11 @@ export function getStyles(props: ITagItemStyleProps): ITagItemStyles {
           ? `${effects.roundedCorner2} 0 0 ${effects.roundedCorner2}`
           : `0 ${effects.roundedCorner2} ${effects.roundedCorner2} 0`,
         selectors: {
+          [`.${classNames.isSelected} &`]: {
+            // selected but not focused, assume color and bg of parent
+            color: 'inherit',
+            background: 'transparent',
```
Comment (smhigley) [inline packages/react/src/components/pickers/TagPicker/TagItem.styles.ts:110]: I'd kinda lean towards making this the same as `:focus`, or just using both selectors for the same style block.

**Review by smhigley — APPROVED** [review]

File: packages/react/src/components/pickers/TagPicker/TagItem.styles.ts [hunk 3]
```diff
@@ -71,7 +65,14 @@ export function getStyles(props: ITagItemStyleProps): ITagItemStyles {
...
       },
-      selected && !disabled && [classNames.isSelected],
+      selected &&
+        !disabled && [
+          classNames.isSelected,
+          {
+            background: palette.themePrimary,
+            color: palette.white,
+          },
+        ],
```
Comment (sopranopillow) [inline packages/react/src/components/pickers/TagPicker/TagItem.styles.ts:75]: Are you supposed to be able to select all the items with ctrl+a when not focusing on one item? Currently you can only select all the items if you are focused on one. Other than that, lgtm.

**Review by sopranopillow — APPROVED** [review]
Minor question

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/BaseButton-mount_base.html">775</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/BaseButton-mount.html">768</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Breadcrumb-mount_base.html">2100</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Breadcrumb-mount.html">2372</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Checkbox-mount_base.html">2067</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Checkbox-mount.html">2093</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/CheckboxBase-mount_base.html">1795</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/CheckboxBase-mount.html">1957</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/ChoiceGroup-mount_base.html">3931</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/ChoiceGroup-mount.html">3984</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/ComboBox-mount_base.html">799</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/ComboBox-mount.html">844</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/CommandBar-mount_base.html">8849</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/CommandBar-mount.html">8835</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/ContextualMenu-mount_base.html">10210</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/ContextualMenu-mount.html">9413</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/DefaultButton-mount_base.html">909</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/DefaultButton-mount.html">910</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/DetailsRow-mount_base.html">3144</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/DetailsRow-mount.html">3220</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/DetailsRowFast-mount_base.html">3238</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/DetailsRowFast-mount.html">3149</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/DetailsRowNoStyles-mount_base.html">3032</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/DetailsRowNoStyles-mount.html">2988</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Dialog-mount_base.html">2313</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Dialog-mount.html">2417</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/DocumentCardTitle-mount_base.html">144</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/DocumentCardTitle-mount.html">135</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Dropdown-mount_base.html">2812</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Dropdown-mount.html">2702</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/FocusTrapZone-mount_base.html">1670</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/FocusTrapZone-mount.html">1514</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/FocusZone-mount_base.html">1558</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/FocusZone-mount.html">1449</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/IconButton-mount_base.html">1512</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/IconButton-mount.html">1548</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Label-mount_base.html">265</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Label-mount.html">278</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Layer-mount_base.html">2772</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Layer-mount.html">2769</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Link-mount_base.html">411</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Link-mount.html">421</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/MenuButton-mount_base.html">1227</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/MenuButton-mount.html">1170</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/MessageBar-mount_base.html">1886</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/MessageBar-mount.html">1878</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Nav-mount_base.html">2828</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Nav-mount.html">2789</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/OverflowSet-mount_base.html">838</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/OverflowSet-mount.html">926</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Panel-mount_base.html">1882</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Panel-mount.html">1886</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Persona-mount_base.html">821</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Persona-mount.html">767</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Pivot-mount_base.html">1227</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Pivot-mount.html">1104</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/PrimaryButton-mount_base.html">1139</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/PrimaryButton-mount.html">1067</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Rating-mount_base.html">6620</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Rating-mount.html">6391</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/SearchBox-mount_base.html">992</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/SearchBox-mount.html">1036</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Shimmer-mount_base.html">2027</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Shimmer-mount.html">2060</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Slider-mount_base.html">1657</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Slider-mount.html">1639</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/SpinButton-mount_base.html">4191</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/SpinButton-mount.html">4470</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Spinner-mount_base.html">346</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Spinner-mount.html">355</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/SplitButton-mount_base.html">2543</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/SplitButton-mount.html">2613</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Stack-mount_base.html">424</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Stack-mount.html">401</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/StackWithIntrinsicChildren-mount_base.html">1979</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/StackWithIntrinsicChildren-mount.html">1951</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/StackWithTextChildren-mount_base.html">4267</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/StackWithTextChildren-mount.html">4206</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/SwatchColorPicker-mount_base.html">9893</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/SwatchColorPicker-mount.html">9310</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/TagPicker-mount_base.html">2383</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/TagPicker-mount.html">2056</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/TeachingBubble-mount_base.html">77966</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/TeachingBubble-mount.html">79797</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Text-mount_base.html">351</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Text-mount.html">366</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/TextField-mount_base.html">1142</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/TextField-mount.html">1205</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/ThemeProvider-mount_base.html">1079</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/ThemeProvider-mount.html">1069</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/ThemeProvider-virtual-rerender_base.html">638</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/ThemeProvider-virtual-rerender.html">636</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1563</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1541</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Toggle-mount_base.html">685</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/Toggle-mount.html">738</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/buttonNative-mount_base.html">113</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24434/perf-test/buttonNative-mount.html">107</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/components/pickers/TagPicker/TagItem.styles.ts (hunk 3 above)
Comment (micahgodbolt) [reply packages/react/src/components/pickers/TagPicker/TagItem.styles.ts:75]: yeah, that was the old behavior and is the current behavior. I agree it'd be nice to support both in the future.

Comment (micahgodbolt) [reply packages/react/src/components/pickers/TagPicker/TagItem.styles.ts:75]: probably a limitation of how we are using SelectionZone. If focus isn't on a selectable item, ctrl+a doesn't do anything.

**Pull request merged**